pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionConfig.java

```diff
@@ -106,11 +106,11 @@
     private Map<String, Object> userConfig;
     private Runtime runtime;
     private boolean autoAck;
-    private int maxMessageRetries;
```

Comment:
Why do we need to set the defaults here?

Reply:
In general functionconfig(and sourceconfig/sinkconfig) are user facing apis. In that sense, its always good to explicitly set some variables as default(instead of setting them in the code anyways).

Reply:
we also have a method called inferMissingArguments for sources, sinks, and functions in which we try to use default values.  We should have the default values set one one place though.